internal/services/network/subnet_data_source.go

```diff
@@ -58,11 +58,12 @@ func dataSourceSubnet() *pluginsdk.Resource {
 			},
 
 			"service_endpoints": {
-				Type:     pluginsdk.TypeList,
+				Type:     pluginsdk.TypeSet,
```

Comment:
a Data Source cannot be a Set, so this should be a List

```suggestion
				Type:     pluginsdk.TypeList,
```